src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1121,18 +1121,32 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             int typeID = svThisPtr.metaData.type;
 
             //Test for exact match
-            List<FunctionEndPoint> exactFeps = new List<FunctionEndPoint>();
+            List<FunctionEndPoint> exactFeps = null;
 
-            foreach (FunctionEndPoint fep in feps)
-                if (fep.ClassOwnerIndex == typeID)
-                    exactFeps.Add(fep);
+            // Is static method call (i.e no this pointer)
+            if (svThisPtr.Pointer == Constants.kInvalidIndex)
+            {
+                // Here we will cover the specific case of static method hiding.
+                // We do not need to check actually if the method has the "IsHideBySig" (https://docs.microsoft.com/en-us/dotnet/api/system.reflection.methodbase.ishidebysig)
+                // because static methods can only be hidden.
+                // In this case we simply select the function that belongs to the calling class.
+                // We also need to check that all function end points in "feps" have the exact same signature.
+                var sig = feps.First().Signature(false/*includeReturnType*/);
+                if (feps.All(x => x.Signature() == sig))
+                {
+                    exactFeps = feps.Where(x => x.ClassOwnerIndex == stackFrame.ClassScope).ToList();
+                }
+            } else
```

Comment:
weird formatting I think.

Reply:
What do you mean ? 😕 
The comments ?

Reply:
Just meant the `} else` on the same line - not a big deal - if you format the document just curious if it gets moved.

Reply:
format does move the `} else`